apps/nextjs/src/app/api/health/live/route.ts

```diff
@@ -0,0 +1,93 @@
+import { performance } from "perf_hooks";
+
+import { db } from "@homarr/db";
+import { logger } from "@homarr/log";
+import { handshakeAsync } from "@homarr/redis";
+
+export async function GET() {
+  const timeBeforeHealthCheck = performance.now();
+  const response = await executeAndAggregateAllHealthChecksAsync();
+  logger.info(`Completed healthcheck after ${performance.now() - timeBeforeHealthCheck}ms`);
+
+  if (response.status === "healthy") {
+    return new Response(JSON.stringify(response), {
+      status: 200,
+    });
+  }
+
+  return new Response(JSON.stringify(response), {
+    status: 500,
+  });
+}
+
+const executeAndAggregateAllHealthChecksAsync = async (): Promise<{
+  healthChecks: Record<string, object>;
+  status: "healthy" | "unhealthy";
+}> => {
+  const healthChecks = [
+    executeHealthCheckSafelyAsync("database", async () => {
+      const before = performance.now();
+      // sqlite driver does not support raw query execution. this is for a heartbeat check only - it doesn't matter if data is returned or not
+      await db.query.serverSettings.findFirst();
+      const after = performance.now();
+      return {
+        latency: after - before,
+      };
+    }),
+    executeHealthCheckSafelyAsync("redis", async () => {
+      const before = performance.now();
+      await handshakeAsync();
+      const after = performance.now();
+      return {
+        latency: after - before,
+      };
+    }),
+  ];
+
+  const healthCheckResults = await Promise.all(healthChecks);
```

Comment:
Use Promise.allSettled instead, otherwise it will not respond anything when unhealthy

Reply:
``Promise.allSettled`` returns a list of ``PromiseSettledResult<HealthCheckResult>[]``. That's quite inconvinient because I always have to filter for the resolved promises and TypeScript isn't smart enough that iterations will have the type of the resolved promise. Also, since I catch any errors in the safe method, which returns the promise, it should never return a rejected promise, no?
Can you make a suggestion otherwise with ``allSettled`` and suggestions for the above mentioned problems?

Reply:
Okay haven't seen that you have a try catch in there, then it's fine